### DISCUSSION TITLE
VMs are blurry without Retina mode

### DISCUSSION THREAD

**Discussion by LostInCompilation** [utmapp/UTM#4897] [UTM Help (macOS)] [answered]
Every VM is quiet blurry when retina mode is disabled. I tried every scaling option there is with no improvement. It's probably a configuration error on my side. Or is it just expected?
Retina mode is not always an option, since some older systems can't support these high resolutions or become really slow.
Different display devices make no difference.

16" M1 Pro MacBook Pro

**Comment by osy** [comment]
Is this new to 4.1.5 or has it always been the case? If it’s always been the case I think what you see as “blurry” is what’s expected from non-hidpi and you’re just used to it.

**Comment by LostInCompilation** [comment, marked as answer]
This was always the case. I did some experimentation. For example if I install WindowsXP using the config file from the UTM website, which uses the SPICE QXL GPU device, the screen is blurry. Even when I change the display device afterwards. When I install WindowsXP with virtio-vga-gpu it is not blurry.

Here are screenshots comparing the two. Same resolution, no retina mode, running with same display device and VM settings.
Blurry:
<img width="533" alt="blur" src="https://user-images.githubusercontent.com/12819635/210676645-24f32611-8c0b-4f5c-bcfd-8c793dddf3a5.png">

Good:
<img width="599" alt="good" src="https://user-images.githubusercontent.com/12819635/210676648-2a6d6f4d-0168-40b1-afa2-d0a451e068bb.png">

**Reply by osy** [reply]
Perhaps the example config uses a different upscaler/downscaler?

**Reply by LostInCompilation** [reply]
Oh well. I messed up. Somehow I screwed up with the upscaler settings. Thought I tested them all...
Yeah nearest neighbour for upscaler is working fine now.

**Reply by felixlandis** [reply]
"Yeah nearest neighbour for upscaler is working fine now." what does that mean? I have the same issue

**Reply by m-zta** [reply]
Yeah I have the same issue. What the frick do you mean "nearest neighbour for upscaler"??

**Reply by LostInCompilation** [reply]
You can select the upscaler in the VM settings.
<img width="799" alt="UTM" src="https://github.com/utmapp/UTM/assets/12819635/d7c9aa9b-2968-4956-86e6-7354af802dfb">

**Comment by TheRealGanesh** [comment]
@LostInCompilation 
Quick question, I am experiencing the same issue that you experience right now.  I have installed a copy of Ubuntu Server, with a Xubuntu Desktop on it, and it's blurry.  What do you mean when you say you install with virtio-vga-gpu.  Upon using UTM right now, that option is not in my Display dropdown menu at all, and I am not able to select it when I install my OS.  Could you please clarify?
As of now, these are the default settings it installs itself with:
<img width="484" alt="Screenshot 2024-12-28 at 10 16 52 PM" src="https://github.com/user-attachments/assets/aa9b2c6d-ea3f-4ae0-82cb-940776bb6999" />
<img width="665" alt="Screenshot 2024-12-28 at 10 17 00 PM" src="https://github.com/user-attachments/assets/0ebb5beb-b7b1-4b27-a00c-1f249a13de45" />

**Reply by suhaybu** [reply]
Toggling Retina mode on helped me on my ubuntu server. Everything will look small, but to fix that, you can go to your display settings inside Ubuntu and play with the scaling. Hope it helps!
